app/controllers/products_controller.rb

```diff
@@ -1,18 +1,67 @@
 class ProductsController < ApplicationController
+  before_filter :authenticate_user!, except: [ :index, :show ]
+
   def index
     @products = Product.all
     @categories = Category.all
   end
   def show
-        @categories = Category.all
-        @product = Product.find(params[:id])
-        @comments = Comment.where(product_id:params[:id])
+    @categories = Category.all
+    @product = Product.find(params[:id])
+    @designer = User.find(@product.designer_id)
+    @comments = Comment.where(product_id:params[:id])
+    @retailprice = @product.product_details.order("retail_price ASC").first
   end
   def new
+    @product = Product.new
   end
-
   def create
-    # ajout if logged in apres
-   Product.create!(name:params[:name],description:params[:description],image:params[:image])
+    @product = Product.new(name:params[:name],description:params[:description],image:params[:image],designer_id:current_user.id)
+    if @product.save
+      redirect_to action: "edit", id: @product.id
+    else
+      render 'new'
+    end
+     if current_user.artist == false
+       user = User.find(current_user.id)
+       user.update_attribute(:artist, true)
+     end
+  end
+  def edit
+    @product = Product.find(params[:id])
+    if current_user.id == @product.designer_id
+      @attributs = ProductDetail.where(product_id:@product.id)
+    end
+  end
+  def update
+    @product = Product.find(params[:id])
+    if current_user.id == @product.designer_id
+      @attributs = ProductDetail.where(product_id:@product.id)
+      @product.update_attributes(name:params[:name],description:params[:description])
+      @attributs.each do |attribut|
+        unless params[attribut.id.to_s] == nil
+         attribut.update_attributes(designer_markup:params[attribut.id.to_s])
+        end
+      end
+      redirect_to action: "show", id: @product.id
+    end
+  end
+  def manage
+    @products = Product.where(designer_id:current_user.id)
+  end
+  def comission
+    @payed_order = Order.all.select{|x| x.payment.allowed == true}.each{|order| order.order_details.select{|order|  order.product.designer_id == current_user.id }}
```

Comment:
Shadowing outer local variable - `order`.<br>Line is too long. [163/80]<br>Unnecessary spacing detected.<br>Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.